app/src/main/java/com/heig/atmanager/MainActivity.java

```diff
@@ -254,4 +257,63 @@ private void signOut(){
         Intent intent = new Intent(MainActivity.this,SignInActivity.class);
         startActivity(intent);
     }
+
+    /**
+     * Enables the back button instead of the drawer menu
+     *
+     * @param enable : back button status
+     */
+    public void enableBackButton(boolean enable) {
+
+        if(enable) {
+            // Disable slide-to-open drawer navigation
+            drawerLayout.setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED);
+            // Hide drawer icon
+            drawerToggle.setDrawerIndicatorEnabled(false);
+            // Show back arrow icon
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        } else {
+            // Enable slide-to-open drawer navigation
+            drawerLayout.setDrawerLockMode(DrawerLayout.LOCK_MODE_UNLOCKED);
+
+            // Create and sync drawer toggle
+            drawerToggle = new ActionBarDrawerToggle (this, drawerLayout, R.string.drawer_open, R.string.drawer_close);
+            drawerLayout.addDrawerListener(drawerToggle);
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+            drawerToggle.syncState();
+        }
+    }
+
+    @Override
+    public void onBackPressed() {
+        displayPreviousFragment(previousFragment);
```

Comment:
It's wiser to pop the call stack ?
Something like this ` getFragmentManager().popBackStack();`